src/main/java/com/faforever/client/map/management/MapsManagementController.java

```diff
@@ -0,0 +1,180 @@
+    this.duplicatedMapNames = clonedList.stream()
+        .filter(map -> map.getVersion() != null && Collections.frequency(mapNames, map.getDisplayName()) > 1)
+        .map(MapBean::getDisplayName).distinct().collect(Collectors.toList());
+    tableView.setItems(this.maps);
+  }
+
+  private void settingChoiceBox() {
+    filterMapsChoiceBox.setConverter(new StringConverter<>() {
+      @Override
+      public String toString(MapFilter object) {
+        return i18n.get(object.getI18n());
+      }
+
+      @Override
+      public MapFilter fromString(String string) {
+        throw new UnsupportedOperationException(); // not required
+      }
+    });
+    filterMapsChoiceBox.getSelectionModel().selectedItemProperty().addListener((obs, oldFilter, newFilter) -> {
+      selectedMaps.clear();
+      maps.setPredicate(getPredicateBy(newFilter));
+      tableView.scrollTo(0);
+    });
+    filterMapsChoiceBox.setItems(FXCollections.observableArrayList(MapFilter.values()));
+    filterMapsChoiceBox.setValue(MapFilter.CUSTOM_MAPS);
+  }
+
+  private Predicate<MapBean> getPredicateByCurrentFilter() {
+    return getPredicateBy(filterMapsChoiceBox.getValue());
+  }
+
+  private Predicate<MapBean> getPredicateBy(MapFilter filter) {
+    return switch (filter) {
+      case OFFICIAL_MAPS -> mapService::isOfficialMap;
+      case CUSTOM_MAPS -> mapService::isCustomMap;
+      case ALL_MAPS -> null;
+    };
+  }
+
+  private void settingTableView() {
+    tableView.setSelectionModel(null);
+    tableView.setRowFactory(param -> new MapBeanRow());
+
+    previewMapColumn.setCellFactory(param -> new PreviewMapCell());
+    previewMapColumn.setCellValueFactory(param -> new SimpleObjectProperty<>
+        (mapService.loadPreview(param.getValue().getFolderName(), PreviewSize.SMALL)));
+
+    mapNameColumn.setCellFactory(param -> new StringCell<>(item -> {
+      String name = item.getDisplayName();
+      ComparableVersion v = item.getVersion();
+      return v != null && duplicatedMapNames.contains(name)
+          ? String.format("%s | %s %s", name, i18n.get("map.version"), v.toString())
+          : name;
+    }));
+    mapNameColumn.setCellValueFactory(param -> new SimpleObjectProperty<>(param.getValue()));
+
+    checkboxColumn.setCellFactory(param -> new CheckBoxCell(selectedMaps, mapService));
+    checkboxColumn.setCellValueFactory(param -> new SimpleObjectProperty<>(param.getValue()));
+  }
+
+  public void onRemoveButtonClicked() {
+    CompletableFuture.runAsync( () -> {
+      Iterator<MapBean> iterator = selectedMaps.iterator();
+      while (iterator.hasNext()) {
+        MapBean map = iterator.next();
+        mapService.uninstallMap(map).whenComplete((unused, throwable) -> {
+          if (throwable == null) {
+            iterator.remove();
+          } else {
+            throw new UninstallMapException(String.format("error when uninstall map '%s'", map), throwable);
+          }
+        }).join();
```

Comment:
We don't really want to make this a blocking process as that defeats the purpose of using the CompletableFuture. It looks like you want to make sure all of these tasks finish so that you can update the table. This can be done either by using a completeableFuture.allOf on the list of removal futures. Additionally if the installedMaps list backs the filtered maps list itself as mentioned above it should update automatically